Scripts/Fan10.swift

```diff
@@ -0,0 +1,111 @@
+import ArgumentParser
+
+import SwiftFusion
+import BeeDataset
+import BeeTracking
+import TensorFlow
+import PythonKit
+import Foundation
+
+typealias LikelihoodModel = TrackingLikelihoodModel<DenseRAE, MultivariateGaussian, GaussianNB>
+
+/// Fan10: RP Tracker, using the new tracking model
+struct Fan10: ParsableCommand {
+  @Option(help: "Run on track number x")
+  var trackId: Int = 0
+
+  @Option(help: "Run for number of frames")
+  var trackLength: Int = 80
+
+  @Option(help: "Size of feature space")
+  var featureSize: Int = 30
+
+  @Flag(help: "Training mode")
+  var training: Bool = false
+
+  func getTrainingDataEM(
+    from dataset: OISTBeeVideo,
+    numberForeground: Int = 3000,
+    numberBackground: Int = 3000
+  ) -> [LikelihoodModel.Datum] {
+    let bgBoxes = dataset.makeBackgroundBoundingBoxes(patchSize: (40, 70), batchSize: numberBackground).map {
+      (frame: $0.frame, type: LikelihoodModel.PatchType.bg, obb: $0.obb)
+    }
+    let fgBoxes = dataset.makeForegroundBoundingBoxes(patchSize: (40, 70), batchSize: numberForeground).map {
+      (frame: $0.frame, type: LikelihoodModel.PatchType.fg, obb: $0.obb)
+    }
+    
+    return fgBoxes + bgBoxes
+  }
+  
+  // Just runs an RP tracker and saves image to file
+  // Make sure you have a folder `Results/fan10` before running
+  func run() {
+    let kHiddenDimension = 100
+    let np = Python.import("numpy")
+    let dataDir = URL(fileURLWithPath: "./OIST_Data")
+
+    let (imageHeight, imageWidth, imageChannels) =
+      (40, 70, 1)
+    
+    var rae = DenseRAE(
+      imageHeight: imageHeight, imageWidth: imageWidth, imageChannels: imageChannels,
+      hiddenDimension: kHiddenDimension, latentDimension: featureSize
+    )
+
+    rae.load(weights: np.load("./oist_rae_weight_\(featureSize).npy", allow_pickle: true))
+
+    let generator = ARC4RandomNumberGenerator(seed: 42)
+    var em = MonteCarloEM<LikelihoodModel>(sourceOfEntropy: generator)
+    
+    let trainingDataset = OISTBeeVideo(directory: dataDir, length: 100)!
+    
+    let trainingData = getTrainingDataEM(from: trainingDataset)
+    
+    let trackingModel = em.run(
```

Comment:
Document! I think what you're trying to do here is to run EM but with a different scheme, where the RAE is not retrained? I think this a good hack for now, but now we have a weird mix between protocols and functions. I think we have to choose one or the other, and I like protocols *much* better. BUT, we have to think about how to allow for multiple EM schemes with the same underlying model.